src/solver/conv_asm_implicit_gemm_wrw_gtc_dynamic_xdlops.cpp

```diff
@@ -544,38 +881,132 @@ ConvAsmImplicitGemmGTCDynamicWrwXdlops::GetSolution(const ConvolutionContext& ct
     result.construction_params.push_back(kernel);
 
     const auto& conv_problem = ctx.conv_problem;
+    const auto& lowp_quant   = ctx.conv_problem.GetConv().lowp_quant;
 
-    result.invoker_factory = [conv_problem,
-                              log2_gemm_k_global_splits](const std::vector<Kernel>& kernels) {
-        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
-            decltype(auto) data_ctx = primitive_parameters.CastTo<conv::WrWInvokeParams>();
-            const auto& tensors     = data_ctx.tensors;
-            std::vector<KernelInvoke> ks;
-            std::transform(kernels.begin(),
-                           kernels.end(),
-                           std::back_inserter(ks),
-                           [&](const Kernel& k_wrw) { return handle.Run(k_wrw); });
-            float elapsed = 0;
-            float zero    = 0.f;
-
-            SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
-            if(handle.IsProfilingEnabled())
-                elapsed += handle.GetKernelTime();
-
-            elapsed += CallImplicitGemmWrwDynamic(handle,
-                                                  conv_problem,
-                                                  tensors.x,
-                                                  tensors.dy,
-                                                  tensors.dw,
-                                                  ks,
-                                                  log2_gemm_k_global_splits);
-            if(handle.IsProfilingEnabled())
-            {
-                handle.ResetKernelTime();
-                handle.AccumKernelTime(elapsed);
-            }
+    auto opShapeArgs = ComputeDynamicIGemmWrwKernelArgs(conv_problem, log2_gemm_k_global_splits);
+
+    if(conv_problem.IsFp32())
+    {
+        result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+                decltype(auto) data_ctx = primitive_parameters.CastTo<conv::WrWInvokeParams>();
```

Comment:
Please rename `data_ctx` to `invoke_params` or to `wrw_invoke_params` or something like that everywhere (no "data" in the name, please).

Reply:
renamed